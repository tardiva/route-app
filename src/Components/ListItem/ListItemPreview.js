import React, {Component} from 'react';
import Icon from 'react-icons-kit';
import {ic_close} from 'react-icons-kit/md/ic_close';

class ListItemPreview extends Component {

  render() {
    return (
      <div className="list-item" style={this.props.style}>
        <div className="list-item__content">{this.props.name}</div>
        <button className="list-item__delete-button">
          <Icon icon={ic_close}/>
        </button>
      </div>)
  }
}

export default ListItemPreview;