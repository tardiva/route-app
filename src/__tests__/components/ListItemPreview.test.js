import React from 'react';
import { shallow } from 'enzyme';
import ListItemPreview from '../../Components/ListItem/ListItemPreview';

it('should render List Item Preview correctly', () => {
  const wrapper = shallow(<ListItemPreview name="Test" style={{width: '300px'}} />);
  expect(wrapper).toMatchSnapshot();
});