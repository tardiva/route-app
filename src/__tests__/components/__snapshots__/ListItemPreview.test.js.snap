// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render List Item Preview correctly 1`] = `
<div
  className="list-item"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <div
    className="list-item__content"
  >
    Test
  </div>
  <button
    className="list-item__delete-button"
  >
    <Icon
      fill="currentColor"
      icon={
        Object {
          "children": Array [
            Object {
              "attribs": Object {
                "d": "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z",
              },
              "name": "path",
            },
          ],
          "viewBox": "0 0 24 24",
        }
      }
      size={16}
    />
  </button>
</div>
`;
